### PR TITLE
[MRG] ENH/MAINT: refactor connectivity

### PR DESCRIPTION
**Pull request by dengemann**
@Eric89GXL @agramfort some refactoring, preparing for more to come.

### CODE REVIEW

File: mne/connectivity/spectral.py [hunk 1]
```diff
@@ -413,10 +415,6 @@ def _epoch_spectral_connectivity(data, sig_idx, tmin_idx, tmax_idx, sfreq,
     else:
         psd = None
 
-    # tell the methods that a new epoch starts
-    for method in con_methods:
-        method.start_epoch()
-
```
Comment (dengemann) [inline mne/connectivity/spectral.py]: was meaningless

Comment (agramfort) [inline mne/connectivity/spectral.py]: even if you have RtEpochs?

Comment (dengemann) [inline mne/connectivity/spectral.py]: well not a single estimator implemented this one

Comment (dengemann) [inline mne/connectivity/spectral.py]: see the pass above?

Comment (dengemann) [inline mne/connectivity/spectral.py]: I'm wondering how this would change once we add some pearson-based estimators.

Comment (agramfort) [inline mne/connectivity/spectral.py]: no idea. No really time to look now

Comment (dengemann) [inline mne/connectivity/spectral.py]: I will put it back for safety reasons as it would change the API contract, even if our estimarors don't implement it.
Anyway I realize that for amplitude estimators we would need some pattern in which compute con is called directly in each epoch instead of accumulate but nothing happens in the end, no compute_con, only averaging.

Comment (dengemann) [inline mne/connectivity/spectral.py]: >no idea. No really time to look now

nevermind, thanks it helped clarify things on my end.

**Comment by agramfort** [comment]
making this PR WIP. No rush to merge this AFAIK

**Comment by Eric89GXL** [comment]
Pushing this one to 0.12

**Comment by Eric89GXL** [comment]
Pushing this one to 0.13 @dengemann

**Comment by dengemann** [comment]
ok.

**Comment by Eric89GXL** [comment]
@dengemann do you still need this?

**Comment by dengemann** [comment]
Yes this is still pending. Please don't close.

**Comment by dengemann** [comment]
Ok @Eric89GXL @agramfort this one should be good to go now.

File: mne/connectivity/spectral.py [hunk 2]
```diff
@@ -4,12 +4,12 @@
 
 from functools import partial
 from inspect import getmembers
+from inspect import getargspec
 
 import numpy as np
 from scipy.fftpack import fftfreq
 
 from .utils import check_indices
-from ..fixes import _get_args
```
Comment (Eric89GXL) [inline mne/connectivity/spectral.py]: revert this change, `_get_args` is the right way now

File: mne/connectivity/spectral.py [hunk 3]
```diff
@@ -442,29 +442,32 @@ def _epoch_spectral_connectivity(data, sig_idx, tmin_idx, tmax_idx, sfreq,
 
             for method in con_methods:
                 method.accumulate(con_idx, csd)
-    else:
-        # cwt_morlet mode
-        for i in range(0, n_cons, block_size):
+    elif mode in ('cwt_morlet',):  # reminder to add alternative TFR methods
+        for i_block, i in enumerate(range(0, n_cons, block_size)):
             con_idx = slice(i, i + block_size)
+            # XXX checkout conversion to slices here / index tricks
```
Comment (Eric89GXL) [inline mne/connectivity/spectral.py]: I'm not sure what this comment is supposed to indicate

File: mne/connectivity/spectral.py [hunk 4]
```diff
@@ -613,8 +647,9 @@ def spectral_connectivity(data, method='coh', indices=None, sfreq=2 * np.pi,
         number of time points as stc_*. The array-like object can also
         be a list/generator of array, shape =(n_signals, n_times),
         or a list/generator of SourceEstimate or VolSourceEstimate objects.
-    method : string | list of string
-        Connectivity measure(s) to compute.
+    method : string | list of string | callable
```
Comment (Eric89GXL) [inline mne/connectivity/spectral.py]: I would say let's support internally / privately `callable` but not tell users about it. We don't want to have to explain it / support it if we don't have to.

Comment (dengemann) [inline mne/connectivity/spectral.py]: it was actually already supported.

Comment (dengemann) [inline mne/connectivity/spectral.py]: you suggest me to remove the doc?

File: mne/connectivity/spectral.py (hunk 3 above)
Comment (dengemann) [inline mne/connectivity/spectral.py]: some note from the past for a future version of myself who is as puzzled as you about its meaning. will remove it.

File: mne/connectivity/spectral.py (hunk 2 above)
Comment (dengemann) [inline mne/connectivity/spectral.py]: wait. it seemed unused. you want me to use this for the inspection?

**Comment by Eric89GXL** [comment]
Yes, getargspec does not work on 3.5

**Comment by Eric89GXL** [comment]
Yes please remove the doc

**Comment by dengemann** [comment]
Good to go! Please merge if happy!

File: mne/time_frequency/tfr.py [hunk 5]
```diff
@@ -1155,6 +1155,44 @@ def _onselect(self, eclick, erelease, baseline, mode, layout):
                              title=ch_type, vmin=None, vmax=None,
                              axes=ax)
 
+    def to_evoked(self, freqs='default', summary_function=np.std):
```
Comment (Eric89GXL) [inline mne/time_frequency/tfr.py]: ???

**Comment by dengemann** [comment]
Oh... something Idid not mean to commit.

**Comment by dengemann** [comment]
@Eric89GXL now it should be good.

**Comment by Eric89GXL** [comment]
Thanks @dengemann
